### ISSUE TITLE
pickr.on("change")

### ISSUE THREAD

**Issue by bsastregx** [Simonwep/pickr#193]
Hi!
Is it possible to know if pickr.on("change") was submitted by direct handling the pickr color window, or, on the contrary, by changing the value on the input? 

Thanks!
```
Version (see Pickr.version):
Used bundle (es5 or normal one):
Used theme (default is classic): 
Browser-version:  
Operating-system:  
```

**Comment by Simonwep** [comment]
The `change` event gets fired immediatly if:
1. The user **changed the input-value** and the value is **valid** ([related line in source](https://github.com/Simonwep/pickr/blob/master/src/js/pickr.js#L369)).
2. The function [`addSwatch`](https://github.com/Simonwep/pickr/blob/master/src/js/pickr.js#L593) (e.g. if you're **adding a new swatch color**) is called.
3. The user **used the controls** to manipulate the value wherby [`_updateOutput`](https://github.com/Simonwep/pickr/blob/master/src/js/pickr.js#L495) (internal function) will be called. Not that this won't get called when the user manually edits the value since `_recalc` is `false` in this case to prevent any kind of double-calculation.

**Comment by Simonwep** [comment]
Closed due inactivity.

**Comment by dmarman** [comment]
I think a specific event for input should be fired. Another possibility is to add a `changeSource` item in the `instance` object  with the source that fired the event like: `changeSource: 'input'`, `changeSource: 'preview'`.

**Why?**
The `'change'` event coming from the preview fires many times per second. This doesn't allow to add an event listener that takes long to run >30ms, otherwise the preview gets blocked. So any heavy calculation has to go to `'changestop'` event and only light calculations on `'change'`.
The problem is that `'changestop'` won't fire from the input, so I cannot fire any heavy calculation when the user changes the input.

**Possible solutions**
- Fire also `'changestop'` after user changes input.
- Add item on the `instance` object with the change source: input or preview.
- Add specific event for input

**Comment by Simonwep** [comment]
@dmarman The only thing I'd do is fire `changestop` after the user edited the color manually using the input field. Adding a changeSource property would raise confusion and seems to be too case-specific.

@bsastregx I could possibly add a `source` property to both events for where the events came from.

**Comment by Simonwep** [comment]
Fixed in a05661631faa71d6d2738de78f3dfc4eea834eb8. I'll publish a new version today.
